service/mapper/transitions.go

```diff
@@ -326,65 +304,32 @@ func (t *Transitions) IndexChain(s *State) error {
 	return nil
 }
 
-// UpdateTree updates the state's tree. If the state's forest already matches with the next block's state commitment,
-// it immediately returns and sets the state's status to StatusMatched.
+// UpdateTree gets all trie updates and stores it in the state
 func (t *Transitions) UpdateTree(s *State) error {
 	if s.status != StatusUpdate {
 		return fmt.Errorf("invalid status for updating tree (%s)", s.status)
 	}
-
-	log := t.log.With().Uint64("height", s.height).Hex("last", s.last[:]).Hex("next", s.next[:]).Logger()
-
-	// If the forest contains a tree for the commit of the next finalized block,
-	// we have reached our goal, and we can go to the next step in order to
-	// collect the register payloads we want to index for that block.
-	ok := s.forest.Has(s.next)
-	if ok {
-		log.Info().Hex("commit", s.next[:]).Msg("matched commit of finalized block")
-		s.status = StatusCollect
-		return nil
-	}
-
-	// First, we get the next tree update from the feeder. We can skip it if
-	// it doesn't have any updated paths, or if we can't find the tree to apply
-	// it to in the forest. This usually means that it was meant for a pruned
-	// branch of the execution forest.
-	update, err := t.feed.Update()
+	log := t.log.With().Uint64("height", s.height).Logger()
+	// grab updates and move on to next state
+	updates, err := t.feed.Updates()
 	if errors.Is(err, archive.ErrUnavailable) {
 		time.Sleep(t.cfg.WaitInterval)
 		log.Debug().Msg("waiting for next trie update")
 		return nil
 	}
 	if err != nil {
-		return fmt.Errorf("could not feed update: %w", err)
-	}
-	parent := flow.StateCommitment(update.RootHash)
-	tree, ok := s.forest.Tree(parent)
-	if !ok {
-		log.Warn().Msg("state commitment mismatch, retrieving next trie update")
-		return nil
-	}
-
-	// We then apply the update to the relevant tree, as retrieved from the
-	// forest, and save the updated tree in the forest. If the tree is not new,
-	// we should error, as that should not happen.
-	paths, payloads := pathsPayloads(update)
-	tree, _, err = trie.NewTrieWithUpdatedRegisters(tree, paths, payloads, false)
-	if err != nil {
-		return fmt.Errorf("could not update tree: %w", err)
+		return fmt.Errorf("unable to retrieve trie updates for block height %x", s.height)
 	}
-	s.forest.Save(tree, paths, parent)
-
-	hash := tree.RootHash()
-	log.Info().Hex("commit", hash[:]).Int("registers", len(paths)).Msg("updated tree with register payloads")
-
+	s.updates = updates
+	log.Info().Uint64("height", s.height).Msg("Collected Trie Updates to be mapped")
```

Comment:
```suggestion
	log.Info().Msg("Collected Trie Updates to be mapped")
```
log already includes height